### ISSUE TITLE
Mapper Request: MMM01

### ISSUE THREAD

**Issue by ajgowans** [MiSTer-devel/Gameboy_MiSTer#165]
Can the MMM01 mapper be added to the core?

https://gbhwdb.gekkio.fi/cartridges/mmm01.html

It is used for the game: Momotarou Collection 2

**Comment by birdybro** [comment]
Do you have an example of a software emulator which supports MMM01? I looked at the source code for Sameboy and it's not implemented yet.

**Comment by paulb-nl** [comment]
There is a very nice detailed documentation about MMM01 here: https://wiki.tauwasser.eu/view/MMM01

This mapper has been added. It is the only mapper where the header is at the end of ROM because the chip boots up at the last bank. 

Detection is done by checking if there is a Nintendo logo from the header at the end of ROM. MiSTer doesn't provide the file size during download of the ROM so I made it check the end of every 512KB block which is the smallest size of MMM01 ROMs.  I hope there aren't any non-MMM01 ROMs with a Nintendo logo at that position.

The Mani 4-in-1 DMG-602 to 605 ROMs also use this mapper but they set the header to MBC3. Probably on purpose. The mapper type is forced to MMM01 on these.
